lib/travis/api/serialize/v0/pusher/job.rb

```diff
@@ -36,15 +36,18 @@ def job_data(job)
                   'repository_private' => job.repository.private,
                   'build_id' => job.source_id,
                   'commit_id' => job.commit_id,
-                  'log_id' => job.log_id,
                   'number' => job.number,
                   'state' => job.state.to_s,
                   'started_at' => format_date(job.started_at),
                   'finished_at' => format_date(job.finished_at),
                   'queue' => job.queue,
                   'allow_failure' => job.allow_failure,
                   'annotation_ids' => job.annotation_ids
-                }
+                }.tap do |ret|
```

Comment:
Since you are modifying the job models for def log I would suggest also adding this into the job models:
``` ruby 
def log_id
  return super unless Travis.config.logs_api.enabled?
  nil
end
```

you can either return nil or '' in that method if you want to avoid modifying these result objects everywhere.